### ISSUE TITLE
Unmonitored websocket task triggers false errors in logs

### ISSUE THREAD

**Issue by rgeronimi** [fafhrd91/aiohttp#8433]
### Describe the bug

The websocket heartbeat functionnality sends a ping message every N seconds (both on the client and the server side). However this task is not monitored, leading to spurious asyncio error logs when it fails (e.g., because the connection was closed down in parallel by another task), e.g.:
```
2024-05-01 07:29:09,261 ERROR asyncio Task exception was never retrieved
future: <Task finished name='Task-117' coro=<WebSocketWriter.ping() done, defined at /opt/homebrew/Caskroom/miniforge/base/envs/work/lib/python3.11/site-packages/aiohttp/http_websocket.py:711> exception=ConnectionResetError('Cannot write to closing transport')>
```

The code on the server side indeed does not retrieve exceptions thrown out by this task:
https://github.com/aio-libs/aiohttp/blob/f662958b150a9d8d92fcbd0c9235e6bee1bedd67/aiohttp/web_ws.py#L151

Same code on the client side:
https://github.com/aio-libs/aiohttp/blob/f662958b150a9d8d92fcbd0c9235e6bee1bedd67/aiohttp/client_ws.py#L115

In both case, the fix for these 2 lines is simple: just attach a completion callback to the task that retrieves the exception, if any :
```
task = self._loop.create_task(self._writer.ping())
def on_ping_done(future: asyncio.Future[None]) -> None:
    if not future.cancelled():
        future.exception()
task.add_done_callback(on_ping_done)
```

### To Reproduce

Start a web socket client or server, run connections, and break them, in large number, with asyncio's debug mode enabled (`loop.set_debug()`).

### Expected behavior

No error logs

### Logs/tracebacks

```python-traceback
2024-05-01 07:29:09,261 ERROR asyncio Task exception was never retrieved
future: <Task finished name='Task-117' coro=<WebSocketWriter.ping() done, defined at /opt/homebrew/Caskroom/miniforge/base/envs/work/lib/python3.11/site-packages/aiohttp/http_websocket.py:711> exception=ConnectionResetError('Cannot write to closing transport')>
```

### Python Version

```console
$ python --version
Python 3.11.9
```

### aiohttp Version

```console
$ python -m pip show aiohttp
Name: aiohttp
Version: 3.9.5
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: 
Author-email: 
License: Apache 2
Location: /opt/homebrew/Caskroom/miniforge/base/envs/work/lib/python3.11/site-packages
Requires: aiosignal, attrs, frozenlist, multidict, yarl
Required-by: aiohttp-jinja2, asyncpraw, asyncprawcore, datasets, httpstan, pystan
```

### multidict Version

```console
$ python -m pip show multidict
Name: multidict
Version: 6.0.5
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /opt/homebrew/Caskroom/miniforge/base/envs/work/lib/python3.11/site-packages
Requires: 
Required-by: aiohttp, yarl
```

### yarl Version

```console
$ python -m pip show yarl
Name: yarl
Version: 1.9.4
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache-2.0
Location: /opt/homebrew/Caskroom/miniforge/base/envs/work/lib/python3.11/site-packages
Requires: idna, multidict
Required-by: aiohttp, asyncprawcore
```

### OS

MacOS Sonoma 14.5

### Related component

Server, Client

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by Dreamsorcerer** [comment]
Yeah, this limitation is hinted at by the comment above that line of code. Could certainly be improved.

**Comment by Dreamsorcerer** [comment]
I believe this has been resolved by a recent PR.

**Comment by bdraco** [comment]
Yes. The task should no longer get prematurely garbage collected
